pkg/cmd/get/get_preview_test.go

```diff
@@ -0,0 +1,160 @@
+package get_test
+
+import (
+	"os"
+	"testing"
+
+	"github.com/jenkins-x/jx/pkg/kube"
+
+	"github.com/jenkins-x/jx/pkg/cmd/get"
+	"github.com/jenkins-x/jx/pkg/cmd/opts"
+	"github.com/jenkins-x/jx/pkg/cmd/testhelpers"
+	"github.com/jenkins-x/jx/pkg/gits"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	jenkinsv1 "github.com/jenkins-x/jx/pkg/apis/jenkins.io/v1"
+	clientmocks "github.com/jenkins-x/jx/pkg/cmd/clients/mocks"
+	helm_test "github.com/jenkins-x/jx/pkg/helm/mocks"
+	resources_test "github.com/jenkins-x/jx/pkg/kube/resources/mocks"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestCurrentPreviewUrlWithoutEnvVars(t *testing.T) {
+	ro := os.Getenv("REPO_OWNER")
+	rn := os.Getenv("REPO_NAME")
+	jn := os.Getenv("JOB_NAME")
+
+	defer os.Setenv("REPO_OWNER", ro)
+	defer os.Setenv("REPO_NAME", rn)
+	defer os.Setenv("JOB_NAME", jn)
+
+	commonOpts := &opts.CommonOptions{}
+	factory := clientmocks.NewMockFactory()
+	commonOpts.SetFactory(factory)
+
+	options := &get.GetPreviewOptions{
+		GetEnvOptions: get.GetEnvOptions{
+			GetOptions: get.GetOptions{
+				CommonOptions: commonOpts,
+			},
+		},
+	}
+
+	want := "No $JOB_NAME defined for the current pipeline job to use"
+
+	if _, err := options.CurrentPreviewUrl(); err != nil {
+		if err.Error() != want {
+			t.Errorf("CurrentPreviewUrl() failed with wrong error got `%s' want `%s'", err.Error(), want)
+		}
+	} else {
+		t.Errorf("CurrentPreviewUrl() didn't fail, expected error: `%s'", want)
+	}
+}
+
+func TestCurrentPreviewUrlWithoutPreviews(t *testing.T) {
+	os.Setenv("REPO_OWNER", "jx-testing")
+	os.Setenv("REPO_NAME", "jx-testing")
+	defer os.Unsetenv("REPO_OWNER")
+	defer os.Unsetenv("REPO_NAME")
+
+	commonOpts := &opts.CommonOptions{}
+	commonOpts.SetDevNamespace("jx-testing")
+
+	factory := clientmocks.NewMockFactory()
+	user := jenkinsv1.User{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: "jx",
+			Name:      "test-user",
+		},
+		Spec: jenkinsv1.UserDetails{
+			Name:     "Test",
+			Email:    "test@test.com",
+			Accounts: make([]jenkinsv1.AccountReference, 0),
+		},
+	}
+
+	testhelpers.ConfigureTestOptionsWithResources(commonOpts,
+		[]runtime.Object{},
+		[]runtime.Object{&user},
+		&gits.GitFake{CurrentBranch: "job"},
+		&gits.FakeProvider{},
+		helm_test.NewMockHelmer(),
+		resources_test.NewMockInstaller(),
+	)
+
+	commonOpts.SetFactory(factory)
+
+	options := &get.GetPreviewOptions{
+		GetEnvOptions: get.GetEnvOptions{
+			GetOptions: get.GetOptions{
+				CommonOptions: commonOpts,
+			},
+		},
+	}
+
+	want := "No Preview for name: jx-testing-jx-testing-job"
```

Comment:
This is failing because it's still using `$JOB_NAME` behind the scenes - I think you probably want to set that to `job` for this test, reverting it back afterwards.

Reply:
Actually, it may be that you need to unset `JOB_NAME` or set it to `""`...

Reply:
In fact, $JOB_NAME is set in the container running the tests I guess?

Reply:
Yup, hence the test failing with `jx-testing-jx-testing-pr-5824` as the preview name it's looking for.

Reply:
@abayer @dgozalo is $JOB_NAME set in the container only? I think this is part of the problem the user is reporting. How is this value supposed to be set when calling it from the command line from a users' machine which is interacting with a cluster?

Reply:
In that context, `$JOB_NAME` would be unset...

Reply:
@abayer is there a plan to let the user set it via a flag in the cli or some other way? I'm wandering what is the right flow here. E.g. user sees the error:  `error: No $JOB_NAME defined for the current pipeline job to use`. is he now supposed to call `JOB_NAME=x jx get preview --current`? and will the user know what job name stands for?

Reply:
@mauromorales I honestly have no idea. =)

Reply:
I think the ''--current" flag is not supposed to be used from the local machine. This is also implied by the help for this command:

> \# View the current preview environment URL
> \# inside a CI pipeline
> jx get preview --current

Locally we don't have the information we need. It's not just the _JOB_NAME_ which is a problem, but also the repo owner and repo name. We cannot extract them from the environment. You would either have to add explicit options for this or assume that this command gets run from the repository for which you want to retrieve the preview information in which case you could get the repo information from the git info. 

Also, what does 'current' even mean in this context?  The issue is imo not clearly defined and we need to discuss first on the issue what the path forward is. 

The '--current' flag is unfortunate imo, since it does not fit the semantics of the command. `jx get previews` is intended to be independent of the directory you run it from. It accesses the env settings in the cluster. A '--current' flag needs more context and would need to be run from an actual project directory.

I'd remove the whole flag. This might be a bit dangerous in case this flag is used in some existing build packs of ours. If we need to keep it, we could check whether we are in Cluster and if not return a warning to the user that this option is not available locally.

Reply:
I added several comments on the issue https://github.com/jenkins-x/jx/issues/3713 and removed the "good first issue" label. This is not trivial at all.

Reply:
Makes sense, let's have the conversation on the issue then and I'll focus on having the tests just reflect the current implementation